src/libAtomVM/bitstring.c

```diff
@@ -153,7 +153,7 @@ enum UnicodeTransformDecodeResult bitstring_utf8_decode(const uint8_t *buf, size
         v |= (buf[3] & 0x3F);
         // overlong encoding or invalid codepoint
         if (v <= 0x10000 || v > 0x10FFFF) {
-            return false;
+            return UnicodeTransformDecodeFail;
```

Comment:
is this a bugfix over the existing codebase?

Reply:
Yes, and it affects the behavior of this PR.  See the discussion in the Telegram channel.

Reply:
I see.
Opening a different PR might be too much, but I always suggest separating bug fixes, refactors and new features in different commits.
IMHO priority rule should be: bug fixing, refactoring and top of that new feature.

Reply:
I understand.